### DISCUSSION TITLE
Is there a way to clear filerejections array on demand?

### DISCUSSION THREAD

**Discussion by sidx64** [react-dropzone/react-dropzone#1340] [Q&A]
Is there a way to clear filerejections array on demand? 

Let's say user uploads a file and it is too large, so is rejected. Now the filerejections array would have the file item. What is the right way to clear that array without having to click/drag new files into the drop zone?

**Comment by rolandjitsu** [comment]
@sidx64 you would need to manage that state by yourself in your application. The dropzone component/hook state changes only when the user interacts with it (starts another drag'n'drop or uses the file picker window).

**Comment by Gustav-Eikaas** [comment]
@rolandjitsu I have the same issue, could we add some sort of `clearRejectedFiles` function to be returned from the `useDropzone` hook?

Basically I want it to show a message when a file is rejected but I want to clear this after x seconds

**Reply by rolandjitsu** [reply]
This can be managed at the application level. You can wrap the hook in your own hook that exposes the APIs you need to manage the selected/dropped files.
